pkg/controller/workload/reconciler.go

```diff
@@ -124,26 +126,23 @@ func (r *Reconciler) completeReconciliation(ctx context.Context, workload *v1alp
 		workload.Status.ObservedGeneration = workload.Generation
 		updateErr = r.Repo.StatusUpdate(workload)
 		if updateErr != nil {
-			logger.Error(updateErr, "update error")
-			if err == nil {
-				return ctrl.Result{}, fmt.Errorf("update workload status: %w", updateErr)
-			}
+			return ctrl.Result{}, fmt.Errorf("update workload status: %w", updateErr)
```

Comment:
it seems like the behavior changed a bit here now - it seems like we'd lose the change of logging the original `err` in case we had one given that we now return from here regardless of whether `err` was nil or not 🤔  is that right?

Reply:
This was intentional. @squeedee and I decided that it would be okay if we lost the original error. We requeue here and assume the system will sort it self out.

Reply:
got it, makes sense 👍  thx!